livekit-agents/livekit/agents/beta/tools/http/send_http.py

```diff
@@ -0,0 +1,306 @@
+
+    Raises:
+        ValueError: If config is invalid (missing URL, invalid parameters schema)
+
+    Example:
+        ```python
+        from livekit.agents.beta.tools.http import create_http_tool, HTTPToolConfig
+
+        # Minimal usage
+        tool = create_http_tool(HTTPToolConfig(
+            url="https://api.example.com/status"
+        ))
+
+        # With parameters
+        config = HTTPToolConfig(
+            name="create_ticket",
+            description="Create a support ticket",
+            url="https://api.example.com/tickets",
+            method="POST",
+            headers={"Authorization": "Bearer token"},
+            parameters={
+                "type": "object",
+                "properties": {
+                    "title": {"type": "string", "description": "Ticket title"},
+                    "priority": {"type": "string", "enum": ["low", "high"]}
+                },
+                "required": ["title"]
+            }
+        )
+        tool = create_http_tool(config)
+
+        # With normalizer for custom response handling
+        config_with_norm = HTTPToolConfig(
+            name="get_weather",
+            url="https://api.weather.com/current",
+            method="GET",
+            parameters={
+                "type": "object",
+                "properties": {
+                    "city": {"type": "string"}
+                },
+                "required": ["city"]
+            },
+            output_normalizer=lambda resp: f"Temp: {resp.body}" if resp.success else None
+        )
+        weather_tool = create_http_tool(config_with_norm)
+
+        # Use in agent
+        agent = Agent(tools=[tool, weather_tool])
+        ```
+    """
+    if not config.url:
+        logger.error("HTTP tool creation failed: missing URL")
+        raise ValueError("URL is required in HTTPToolConfig")
+
+    supported_methods = {"GET", "POST", "PUT", "PATCH", "DELETE"}
+    method_upper = config.method.upper()
+    if method_upper not in supported_methods:
+        logger.error(
+            "HTTP tool creation failed: unsupported method",
+            extra={"method": config.method, "supported_methods": list(supported_methods)},
+        )
+        raise ValueError(
+            f"Unsupported HTTP method: {config.method}. "
+            f"Supported methods: {', '.join(sorted(supported_methods))}"
+        )
+
+    parameters_schema = config.parameters or {"type": "object", "properties": {}, "required": []}
+
+    if not isinstance(parameters_schema, dict) or "type" not in parameters_schema:
+        logger.error(
+            "HTTP tool creation failed: invalid parameters schema",
+            extra={"schema_type": type(parameters_schema).__name__},
+        )
+        raise ValueError("Invalid parameters schema - must be a dict with 'type' field")
+
+    tool_name = config.name or "http_request"
+    tool_description = config.description or f"Make {config.method} request to {config.url}"
```

Comment:
i don't think a default description makes sense - if the dev leaves it blank, it should remain blank